docs/data-cookbooks/run-tasks-in-lambda-or-docker.md

```diff
@@ -0,0 +1,110 @@
+
+```yaml
+QueueGranules:
+  handler: index.handler
+  timeout: 900
+  memory: 3008
+  source: node_modules/@cumulus/queue-granules/dist/
+  useMessageAdapter: true
+```
+
+Given the `QueueGranules` lambda package is already used in a Step Function State Machine definition, the following will exist in a workflow YAML file:
+
+```yaml
+    QueueGranules:
+      Type: Task
+      Resource: ${QueueGranulesLambdaFunction.Arn}
+      Next: StopStatus
+```
+
+Given it has been discovered this task can no longer run in AWS Lambda, it can be run on the Cumulus ECS cluster by including the following in `app/config.yml`:
+
+```yaml
+  ecs:
+    instanceType: t2.large
+    desiredInstances: 1
+    services:
+      QueueGranules:
+        docker: true
+        image: cumuluss/cumulus-ecs-task:1.2.5
+        memory: 4000
+        count: 1
+        envs:
+          AWS_DEFAULT_REGION:
+            function: Fn::Sub
+            value: '${AWS::Region}'
+        commands:
+          - cumulus-ecs-task
+          - '--activityArn'
+          - function: Ref
+            value: QueueGranulesActivity
+          - '--lambdaArn'
+          - function: Ref
+            value: QueueGranulesLambdaFunction
+
+  activities:
+    - name: QueueGranules
+```
+
+and then modifying the corresponding Step Function State Machine definition in a workflow YAML file:
+
+```yaml
+    QueueGranules:
+      Type: Task
+      Resource: ${QueueGranulesActivity}
+      Next: StopStatus
+```
+
+## How do I name my resources? A Short Primer on Cumulus Resource Naming Conventions
+
+Cumulus deployments currently depend on the [`kes`](https://github.com/developmentseed/kes) cli tool. `kes` references values in `app/config.yml`, `iam/config.yml`, lambda and workflow `.yml` files to populate the cloudformation templates that are a part of the `@cumulus/deployment` package. `kes` generates final cloudformation files and uploads them to AWS Cloudformation, creating or updating AWS Cloudformation Stacks.
+
+When deploying AWS Lambdas and AWS Activities as detailed above, note the following naming conventions:
+
+* `QueueGranules` in `lambdas.yml`: When kes generates the final cloudformation file, it will define a resource called `QueueGranulesLambdaFunction`. A valid lambda function resource name, such as `QueueGranulesLambdaFunction`, must be used in `app/config.yml` when passed as a command argument to `cumulus-ecs-task`.
+* `QueueGranules` in `app/config.yml`: The string `QueueGranules` exists twice in the `app/config.yml` above.
+    * The first occurence is as a key under `services`. This key provides a descriptive prefix when naming the corresponding ECS Service. It will be included in final cloudformation yaml as an `AWS::ECS::Service` resource with the name `QueueGranulesECSService`. This ECS Service name prefix (`QueueGranules` in this example) can be anything since the service is not referenced in any other part of the deployment at this time.
```

Comment:
```suggestion
    * The first occurrence is as a key under `services`. This key provides a descriptive prefix when naming the corresponding ECS Service. It will be included in final cloudformation YAML as an `AWS::ECS::Service` resource with the name `QueueGranulesECSService`. This ECS Service name prefix (`QueueGranules` in this example) can be anything since the service is not referenced in any other part of the deployment at this time.
```